server/eventlog_test.go

```diff
@@ -633,18 +669,71 @@ func generateNonCosCelEvent(hashAlgoList []crypto.Hash) (cel.Record, error) {
 	return randRecord, nil
 }
 
+func TestParseLinuxKernelState(t *testing.T) {
+	logs := []struct {
+		eventLog
+		name            string
+		expectedCmdline string
+	}{
+		{COS85AmdSev, "COS85AmdSev", test.Cos85AmdSevCmdline},
+		{COS93AmdSev, "COS93AmdSev", test.Cos93AmdSevCmdline},
+		{COS101AmdSev, "COS101AmdSev", test.Cos101AmdSevCmdline},
+	}
+	for _, log := range logs {
+		for _, bank := range log.Banks {
+			hashName := pb.HashAlgo_name[int32(bank.Hash)]
+			subtestName := fmt.Sprintf("%s-%s", log.name, hashName)
+			t.Run(subtestName, func(t *testing.T) {
+				msState, err := parsePCClientEventLog(log.RawLog, bank, GRUB)
+				if err != nil {
+					t.Errorf("failed to parse and replay log: %v", err)
+				}
+
+				if msState.LinuxKernel == nil || len(msState.LinuxKernel.CommandLine) == 0 {
+					t.Errorf("expected %s to have a LinuxKernelState", log.name)
+				}
+
+				if msState.LinuxKernel.CommandLine != log.expectedCmdline {
+					got := len(msState.LinuxKernel.CommandLine)
+					want := len(log.expectedCmdline)
+					if got != want {
+						t.Log(got)
+						t.Log(want)
+					}
+					for i := 0; i < got; i++ {
+						gotAt := msState.LinuxKernel.CommandLine[i]
+						wantAt := log.expectedCmdline[i]
+						if gotAt != wantAt {
+							t.Log(i)
+							t.Logf("%c", gotAt)
+							t.Logf("%c", wantAt)
+							t.Log(gotAt)
+							t.Log(wantAt)
+						}
```

Comment:
IIUC, why need to print the commandline char by char? Also if `len(log.expectedCmdline) < len(msState.LinuxKernel.CommandLine)` line 705 can have  a out of bound err

Reply:
Debug printlines. Removed